File: server/models/chronicle.py
"""编年史模型"""
from pydantic import BaseModel
from typing import Any


class ChronicleLogEntry(BaseModel):
    entryId: str
    playerId: str
    characterName: str
    worldDay: int
    localDay: int
    location: dict
    action: dict
    narrativeOutput: str = ""
    consequences: dict = {}
    timestamp: str = ""


class ChronicleLogBatch(BaseModel):
    playerId: str
    entries: list[ChronicleLogEntry]
    lastWorldDay: int


class PushResult(BaseModel):
    uploaded: int
    failed: int
    newEncounters: list[dict] = []


class WorldChronicleEntry(BaseModel):
    id: str
    worldDay: int
    region: str
    title: str
    narrative: str
    timestamp: str = ""

File: client/src/types/chronicle.ts
export interface ChronicleLogEntry {
  entryId: string;
  playerId: string;
  characterName: string;
  worldDay: number;
  localDay: number;
  location: {
    region: string;
    subRegion: string;
    coordinates: { x: number; y: number; z: number };
  };
  action: {
    summary: string;
    playerChoice: string;
    wasCustomInput: boolean;
    absurdityLevel: number;
    difficulty: number;
    rollResult: string;
    rollDetail: {
      dice: number[];
      modifier: number;
      total: number;
      dc: number;
    };
  };
  narrativeOutput: string;
  consequences: Record<string, unknown>;
  timestamp: string;
  syncStatus: 'pending' | 'synced' | 'failed';
}

export interface ChronicleLogBatch {
  playerId: string;
  entries: ChronicleLogEntry[];
  lastWorldDay: number;
}
